test/unit/presubmit-tests.sh

```diff
@@ -0,0 +1,281 @@
+  mock_presubmit_runners
+  unset pre_build_tests
+  unset post_build_tests
+  unset pre_unit_tests
+  unset post_unit_tests
+  unset pre_integration_tests
+  unset post_integration_tests
+  function check_results() {
+    (( ! PRE_BUILD_TESTS )) || test_failed "Pre build tests did run"
+    (( RAN_BUILD_TESTS )) || test_failed "Build tests did not run"
+    (( ! POST_BUILD_TESTS )) || test_failed "Post build tests did run"
+    (( ! PRE_UNIT_TESTS )) || test_failed "Pre unit tests did run"
+    (( RAN_UNIT_TESTS )) || test_failed "Unit tests did not run"
+    (( ! POST_UNIT_TESTS )) || test_failed "Post unit tests did run"
+    (( ! PRE_INTEGRATION_TESTS )) || test_failed "Pre integration tests did run"
+    (( RAN_INTEGRATION_TESTS )) || test_failed "Custom integration tests did not run"
+    (( ! POST_INTEGRATION_TESTS )) || test_failed "Post integration tests did run"
+    echo "Test passed"
+  }
+}
+
+function test_custom_runners_fail_slow() {
+  mock_presubmit_runners ${FAILURE}
+  unset pre_build_tests
+  unset post_build_tests
+  unset pre_unit_tests
+  unset post_unit_tests
+  unset pre_integration_tests
+  unset post_integration_tests
+  function check_results() {
+    (( ! PRE_BUILD_TESTS )) || test_failed "Pre build tests did run"
+    (( RAN_BUILD_TESTS )) || test_failed "Build tests did not run"
+    (( ! POST_BUILD_TESTS )) || test_failed "Post build tests did run"
+    (( ! PRE_UNIT_TESTS )) || test_failed "Pre unit tests did run"
+    (( RAN_UNIT_TESTS )) || test_failed "Unit tests did not run"
+    (( ! POST_UNIT_TESTS )) || test_failed "Post unit tests did run"
+    (( ! PRE_INTEGRATION_TESTS )) || test_failed "Pre integration tests did run"
+    (( RAN_INTEGRATION_TESTS )) || test_failed "Custom integration tests did not run"
+    (( ! POST_INTEGRATION_TESTS )) || test_failed "Post integration tests did run"
+    echo "Test failed slow"
+  }
+}
+
+function test_custom_runners_fail_fast() {
+  PRESUBMIT_TEST_FAIL_FAST=1
+  mock_presubmit_runners ${FAILURE}
+  unset pre_build_tests
+  unset post_build_tests
+  unset pre_unit_tests
+  unset post_unit_tests
+  unset pre_integration_tests
+  unset post_integration_tests
+  function check_results() {
+    (( ! PRE_BUILD_TESTS )) || test_failed "Pre build tests did run"
+    (( RAN_BUILD_TESTS )) || test_failed "Build tests did not run"
+    (( ! POST_BUILD_TESTS )) || test_failed "Post build tests did run"
+    (( ! PRE_UNIT_TESTS )) || test_failed "Pre unit tests did run"
+    (( ! RAN_UNIT_TESTS )) || test_failed "Unit tests did run"
+    (( ! POST_UNIT_TESTS )) || test_failed "Post unit tests did run"
+    (( ! PRE_INTEGRATION_TESTS )) || test_failed "Pre integration tests did run"
+    (( ! RAN_INTEGRATION_TESTS )) || test_failed "Custom integration tests did run"
+    (( ! POST_INTEGRATION_TESTS )) || test_failed "Post integration tests did run"
+    echo "Test failed fast"
+  }
+}
+
+function test_exempt_md() {
+  source $(dirname $0)/../../scripts/presubmit-tests.sh
```

Comment:
This is no longer in the scripts dir.

Search the rest of the code for scripts?

Reply:
Done.